### PR TITLE
[SPARK-24398] [SQL] Improve SQLAppStatusListener.aggregateMetrics() too slow

### PR DESCRIPTION
**Pull request by frb502**
## What changes were proposed in this pull request?

JIRA Issue: https://issues.apache.org/jira/browse/SPARK-24398
## How was this patch tested?

(Please explain how this patch was tested. E.g. unit tests, integration tests, manual tests)
(If this patch involves UI changes, please attach a screenshot; otherwise, remove this)

Please review http://spark.apache.org/contributing.html before opening a pull request.

### CODE REVIEW

**Comment by felixcheung** [comment]
please see http://spark.apache.org/contributing.html on "Pull Request"

also fix the PR title to start with `[SPARK-24398]`

**Comment by felixcheung** [comment]
Please update the PR description.
Also do you mean "show" -> "slow"?

File: sql/core/src/main/scala/org/apache/spark/sql/execution/ui/SQLAppStatusListener.scala [hunk 1]
```diff
@@ -159,7 +159,7 @@ class SQLAppStatusListener(
   }
 
   private def aggregateMetrics(exec: LiveExecutionData): Map[Long, String] = {
-    val metricIds = exec.metrics.map(_.accumulatorId).sorted
+    val metricIds = exec.metrics.map(_.accumulatorId).toSet
```
Comment (gatorsmile) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/ui/SQLAppStatusListener.scala]: I think we can get rid of `metricIds `

**Comment by gatorsmile** [comment]
ok to test

File: sql/core/src/main/scala/org/apache/spark/sql/execution/ui/SQLAppStatusListener.scala (hunk 1 above)
Comment (gengliangwang) [reply sql/core/src/main/scala/org/apache/spark/sql/execution/ui/SQLAppStatusListener.scala]: +1， we can use `metricTypes`

**Comment by gatorsmile** [comment]
@felixcheung `metricIds` is not needed. We can get rid of it by using `metricTypes `

**Comment by srowen** [comment]
Just following up on this old small PR @gatorsmile @frb502  -- was there a different solution here that makes sense? should this be closed?

**Comment by srowen** [comment]
Resolved by https://issues.apache.org/jira/browse/SPARK-26003

**Pull request closed without merging**